### PR TITLE
Add thumbprint verify

### PR DESCRIPTION
**Pull request by TaoZou1**
Add thumbprint validation, thumbprint number should be 1 or equal to endpoint number if want to verify it.
When tls dial, it will calculate the thumbprint and compare it with local thumbprint.

### CODE REVIEW

File: pkg/nsx/cluster.go [hunk 1]
```diff
@@ -89,14 +93,65 @@ func (cluster *Cluster) NewRestConnector() (*policyclient.RestConnector, *Header
...
+		if tpCount > 1 {
+			for index, ep := range cluster.endpoints {
+				if ep.Host() == host {
+					thumbprint = cluster.config.Thumbprint[index]
+					break
+				}
+			}
+		}
+		config := &tls.Config{
+			InsecureSkipVerify: true,
```
Comment (dantingl) [inline pkg/nsx/cluster.go:112]: Do we need to set `InsecureSkipVerify` as false when thumbprint is there

File: pkg/nsx/cluster.go [hunk 2]
```diff
@@ -89,14 +93,65 @@ func (cluster *Cluster) NewRestConnector() (*policyclient.RestConnector, *Header
...
+		host := strings.Split(addr, ":")[0]
+		var thumbprint string
+		tpCount := len(cluster.config.Thumbprint)
+		if tpCount == 1 {
+			thumbprint = cluster.config.Thumbprint[0]
+		}
+		if tpCount > 1 {
+			for index, ep := range cluster.endpoints {
+				if ep.Host() == host {
+					thumbprint = cluster.config.Thumbprint[index]
```
Comment (dantingl) [inline pkg/nsx/cluster.go:106]: thumbprint and endpoints are matched in order, right?

Comment (TaoZou1) [reply pkg/nsx/cluster.go:106]: yes

File: pkg/nsx/cluster.go (hunk 1 above)
Comment (TaoZou1) [reply pkg/nsx/cluster.go:112]: no, even we verify the thumbprint, the InsecureSkipVerify is true, that verify here is provided by http lib, not include function we added.

**Review by dantingl — APPROVED** [review]

**Pull request merged**